File: UVa/10443.cpp
#include <cstdio>
#include <algorithm>

int lose(char a, char b) {
    if (a == 'S' && b == 'R')
        return 1;
    if (a == 'R' && b == 'P')
        return 1;
    if (a == 'P' && b == 'S')
        return 1;
    return 0;
}


int main() {
    int n = 0;
    scanf("%d", &n);

    for(int test=0;test<n;test++) {
        int x, y, days;
        scanf("%d %d %d", &x, &y, &days);
        char map[100][100];
        char tmp[100][100];
        char s[100];

        for(int j=0;j<x;j++) {
            scanf("%s", map[j]);
        }

        if (test > 0 && test + 1 <= n) {
            printf("\n");
        }

        int d = 0;
        while (d < days) {
            for(int row=0;row<x;row++) {
                for(int column=0;column<y;column++) {
                    char c = map[row][column];

                    if (row+1 < x && lose(map[row][column], map[row+1][column]))
                        c = map[row+1][column];
                    if (row-1 >= 0 && lose(map[row][column], map[row-1][column]))
                        c = map[row-1][column];
                    if (column+1 < y && lose(map[row][column], map[row][column+1]))
                        c = map[row][column+1];
                    if (column-1 >= 0 && lose(map[row][column], map[row][column-1]))
                        c = map[row][column-1];
                    tmp[row][column] = c;
                }
                // printf("TMP:\n");
                // for(int column=0;column<y;column++) {
                //     printf("%c", tmp[row][column]);
                // }
                // printf("\n");
            }
            std::swap(map, tmp);
            d++;
        }
        //printf("FIN\n\n\n\n");
        for(int row=0;row<x;row++) {
            for(int column=0;column<y;column++) {
                printf("%c", map[row][column]);
            }
            printf("\n");
        }
    }
}

File: UVa/10443.py
n = input()
n = int(n)

def lose(a, b):
    #print("Trying ", a, b)
    if a == 'S' and b == 'R':
        return True
    if a == 'R' and b == 'P':
        return True
    if a == 'P' and b == 'S':
        return True
    return False

for i in range(n):
    l = input().split(" ")

    a, b, day = l
    a = int(a)
    day = int(day)
    l = []
    
    for _ in range(a):
        l.append(input())
    
    d = 0
    while d < day:
        tmp = []

        for row in range(len(l)):
            s = []
            for column in range(len(l[row])):
                c = l[row][column]
                
                if row+1 < len(l) and lose(l[row][column], l[row+1][column]):
                    c =  l[row+1][column]
                if row-1 >= 0 and lose(l[row][column], l[row-1][column]):
                    c =  l[row-1][column]
                if column+1 < len(l[row]) and lose(l[row][column], l[row][column+1]):
                    c =  l[row][column+1]
                if column-1 >= 0 and lose(l[row][column], l[row][column-1]):
                    c =  l[row][column-1]
                                    
                s.append(c)
            tmp.append("".join(s))
        tmp, l = l, tmp
        d += 1
    for c in l:
        print(c)
    
    if i + 1 < n:
        print()